### PR TITLE
wrangler: Hyperdrive dev bindings

### PR DESCRIPTION
**Pull request by tmthecoder**
**Associated docs issue(s)/PR(s):**

- N/A

**Author has included the following, where applicable:**

- [x] Tests
- [ ] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

**Note for PR author:**

We want to celebrate and highlight awesome PR review! If you think this PR received a particularly high-caliber review, please assign it the label `highlight pr review` so future reviewers can take inspiration and learn from it.

### CODE REVIEW

File: packages/wrangler/src/config/environment.ts [hunk 1]
```diff
@@ -523,6 +523,8 @@ interface EnvironmentNonInheritable {
 		binding: string;
 		/** The id of the database. */
 		id: string;
+		/** The local postgres connection string for `wrangler dev` */
```
Comment (penalosa) [inline packages/wrangler/src/config/environment.ts:526]: Does this only support postgres?

**Review by penalosa — CHANGES_REQUESTED** [review]
Looking good! This should include some tests, ideally validating that this works end-to-end in local mode (have a look at the e2e folder for some example tests of Wrangler dev—ideally it would be good to validate that a mock postgres server is connected to by `workerd`)

File: packages/wrangler/src/config/environment.ts (hunk 1 above)
Comment (tmthecoder) [reply packages/wrangler/src/config/environment.ts:526]: Currently Hyperdrive only supports postgres, I can change it do `database connection string` for future safety too though

**Comment by tmthecoder** [comment]
@penalosa Should be all set with tests as well. However the socket test won't work until https://github.com/cloudflare/miniflare/pull/727 is merged in & mini flare is bumped here due to a small address formatting bug I found in my initial implementation there

File: packages/wrangler/e2e/dev.test.ts [hunk 2]
```diff
@@ -278,3 +280,91 @@ describe("dev registry", () => {
...
+			expect(url.username).toBe("user");
+			expect(url.password).toBe("pass");
+			expect(url.host).not.toBe("localhost");
+		});
+	});
+
+	it("connects to a socket", async () => {
+		server.on("connection", (sock) => {
+			sock.on("data", (data) => {
+				expect(new TextDecoder().decode(data)).toBe("test string");
```
Comment (penalosa) [inline packages/wrangler/e2e/dev.test.ts:348]: Will this `expect` definitely run? What if the test exits too soon?

Comment (penalosa) [reply packages/wrangler/e2e/dev.test.ts:348]: This may need wrapping in a `new Promise()` that's explicitly await in the test

**Review by penalosa — APPROVED** [review]
LGTM! Let's wait to merge until Miniflare is released and bumped on Monday, so that tests will pass

File: packages/wrangler/e2e/dev.test.ts (hunk 2 above)
Comment (tmthecoder) [reply packages/wrangler/e2e/dev.test.ts]: Ah yeah, that makes sense. Just wrapped it in a promise

**Comment by tmthecoder** [comment]
@penalosa Tests seem to pass locally now that miniflare is the latest in-workspace version. Are we able to rerun the flows here to make sure and then merge?

File: packages/wrangler/e2e/dev.test.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import crypto from "node:crypto";
+import * as nodeNet from "node:net";
 import path from "node:path";
+import { afterEach } from "node:test";
```
Comment (Skye-31) [inline packages/wrangler/e2e/dev.test.ts:4]: This should be from vitest

Comment (tmthecoder) [reply packages/wrangler/e2e/dev.test.ts]: @Skye-31 Ah my bad, it should be changed now. Also fixed the formatting errors that caused tests to fail

**Comment by tmthecoder** [comment]
@rts-rob

**Pull request merged**